core/src/either.rs

```diff
@@ -258,18 +264,27 @@ where
     }
 }
 
+/// Used as an `OutboundSubstream` type, which is the return type for
+/// `open_outbound()`, and an input for `poll_outbound`, both of which
+/// are internal methods of `EitherOutput` (as well as
+/// `destroy_outbound()`, which is wrong API usage).
+/// All calls using `EitherOutbound` dispatch to either `A` or `B`.
 #[derive(Debug, Copy, Clone)]
 #[must_use = "futures do nothing unless polled"]
 pub enum EitherOutbound<A: StreamMuxer, B: StreamMuxer> {
+    /// The first `OutboundSubstream` variant.
     A(A::OutboundSubstream),
```

Comment:
So the panic mentioned above would happen if we ever had `A(B::OutboundSubstream)` which is impossible because of how this type is constructed.

Reply:
Yep, got it now, thanks.